Automattic/simplenote-macos
# Implementing NotesListController: Mark I

### Fix
In this PR we're implementing:

- **NotesListFilter:** Enum that matches the currently selected TagList row
- **NotesListState:** Simple FSM to encapsulate the "Searching" / "Results" states

**PLUS:** As of now, Search Results aren't constrained anymore by the current selected tag.

cc @eshurakov (Thank yooou!!!)
Ref. #626


### Test: Tags List
1. Launch Simplenote / authenticate

- [x] Verify all `All Notes` is preselected by default, and all your notes show up
- [x] Verify that clicking over `Trash` causes the Notes List to display the trashed notes
- [x] Verify that clicking over `Untagged` causes the Notes List to display untagged notes
- [x] Verify that clicking over any `Tag` causes the app to display its contents
- [x] Verify the NotesList Title is always adjusted

### Test: Global Search
1. Click over any `Tag` in your list that contains just a few notes
2. Click over the top right 🔍 icon
3. Enter a `keyword` contained in a note that **does not belong** to any of the notes in your list

- [x] Verify the Search Results presented are global, and aren't constrained anymore by the current tag

### Release
`RELEASE-NOTES.txt` was updated in 6645470 with:

> Search Results are now global #760
